packages/flutter_tools/lib/src/localizations/gen_l10n.dart

```diff
@@ -157,19 +157,19 @@ String generateNumberFormattingLogic(Message message) {
       if (placeholder.hasNumberFormatWithParameters) {
         return numberFormatNamedTemplate
             .replaceAll('@(placeholder)', placeholder.name)
-            .replaceAll('@(format)', placeholder.format)
+            .replaceAll('@(format)', placeholderFormat)
             .replaceAll('@(parameters)', parameters.join(',\n      '));
       } else {
         return numberFormatPositionalTemplate
             .replaceAll('@(placeholder)', placeholder.name)
-            .replaceAll('@(format)', placeholder.format);
+            .replaceAll('@(format)', placeholderFormat);
       }
     });
 
   return formatStatements.isEmpty ? '@(none)' : formatStatements.join('');
 }
 
-String generatePluralMethod(Message message, AppResourceBundle bundle) {
+String _generatePluralMethod(Message message, String translationForMessage) {
```

Comment:
`bundle.translationFor(message)` is nullable, but the call sites make sure `bundle.translationFor(message)` is nonnull before this is called.  Instead of passing in the bundle, pass in the already promoted translation.